app/javascript/lib/imperial.js

```diff
@@ -1834,91 +1869,60 @@ export default class Imperial {
     );
   }
 
-  rondelActions(nation) {
-    const rondelPositions = [
-      'factory',
-      'production1',
-      'maneuver1',
-      'investor',
-      'import',
-      'production2',
-      'maneuver2',
-      'taxation',
-    ];
-    const currentPosition = this.nations.get(nation).rondelPosition;
-    const out = new Set();
-    if (currentPosition) {
-      const currentIndex = rondelPositions.indexOf(currentPosition);
-      const distance = currentIndex - 8;
-      [
-        rondelPositions[currentIndex + 1] || rondelPositions[distance + 1],
-        rondelPositions[currentIndex + 2] || rondelPositions[distance + 2],
-        rondelPositions[currentIndex + 3] || rondelPositions[distance + 3],
-      ].forEach((slot) => {
-        out.add(Action.rondel({ nation, cost: 0, slot }));
-      });
-      out.add(
-        Action.rondel({
-          nation,
-          cost: this.extraRondelCost(1),
-          slot:
-            rondelPositions[currentIndex + 4] || rondelPositions[distance + 4],
-        }),
-      );
-      out.add(
-        Action.rondel({
-          nation,
-          cost: this.extraRondelCost(2),
-          slot:
-            rondelPositions[currentIndex + 5] || rondelPositions[distance + 5],
-        }),
-      );
-      out.add(
+  availableRondelActions(nationName) {
+    const nation = this.nations.get(nationName);
+    const costPerPaidDistance = this.costPerPaidRondelAction(nation);
+
+    const nationCurrentRondelSlot = this.rondel.representationToEntity(nation.rondelPosition);
+    const availableSlots = new AvailableSlots(this.rondel, 3, 3, costPerPaidDistance);
+
+    const availableRondelSlots = new Set();
+
+    const nextAvailableFreeRondelSlots = nationCurrentRondelSlot
+      ? availableSlots.nextAvailableFreeRondelSlots(nationCurrentRondelSlot) : this.rondel.slotOrder;
+    for (const freeRondelSlot of nextAvailableFreeRondelSlots) {
+      availableRondelSlots.add(
         Action.rondel({
-          nation,
-          cost: this.extraRondelCost(3),
-          slot:
-            rondelPositions[currentIndex + 6] || rondelPositions[distance + 6],
+          nation: nationName,
+          cost: 0,
+          slot: freeRondelSlot.representation,
```

Comment:
Returning to the `.representation` vs `.toString` vs other naming question, this is a spot where I feel like `.toString()` would feel pretty natural. Or even doing `freeRondelSlot.name.toString()` would feel natural. Ok I'll drop it now 😄 `representation` is fine, I think I just need to get used to it! 😆

Reply:
I think it's a legitimate concern if the naming feels weird, going to try renaming it to `identifier` for now

It's mostly used to communicate with front-ends/packets, where they're going to refer to the object by ids instead of the pointer

Reply:
[a6e9bfa](https://github.com/Thrillberg/imperial/pull/711/commits/a6e9bfa69e3e1cc1ad7604bf0fa247a5b967bbf0)